test/services/task_list_merge_test.dart

```diff
@@ -0,0 +1,189 @@
+    await Future.delayed(Duration(milliseconds: 10));
+    await devices[1].taskListService.upsertTaskList(taskList);
+    await devices[1].taskListService.upsertTask(taskList.id!, task2);
+
+    await devices[0]
+        .taskListService
+        .mergeCrdtJson(await devices[1].taskListService.crdtToJson());
+    final mergedTaskLists =
+        (await devices[0].taskListService.taskLists).toList();
+    expect(mergedTaskLists.length, 1);
+    final mergedTaskList = mergedTaskLists.first;
+    expect(mergedTaskList.title, taskList.title);
+    expect(mergedTaskList.elements.toSet(), {task2});
+  });
+
+  test('crdt keep more recent task in list', () async {
+    final taskList = TaskList(id: 'id', title: 'list');
+    final task1 = Task(id: 'task1id', title: 'task1');
+    final task2 = Task(id: task1.id, title: 'task2');
+    await devices[1].taskListService.upsertTaskList(taskList);
+    await devices[1].taskListService.upsertTask(taskList.id!, task2);
+    await Future.delayed(Duration(milliseconds: 10));
+    await devices[0].taskListService.upsertTaskList(taskList);
+    await devices[0].taskListService.upsertTask(taskList.id!, task1);
+
+    await devices[0]
+        .taskListService
+        .mergeCrdtJson(await devices[1].taskListService.crdtToJson());
+    final mergedTaskLists =
+        (await devices[0].taskListService.taskLists).toList();
+    expect(mergedTaskLists.length, 1);
+    final mergedTaskList = mergedTaskLists.first;
+    expect(mergedTaskList.title, taskList.title);
+    expect(mergedTaskList.elements.toSet(), {task1});
+  });
+
+  test('crdt recursive task merge in list', () async {
+    final taskList = TaskList(id: 'id', title: 'list');
+    final task1 = Task(
+      id: 'task1Id',
+      title: 'task1',
+      description: 'description1',
+    );
+    final task2 = Task(
+      id: task1.id,
+      title: 'task2',
+      description: 'description2',
+    );
+    await devices[0].taskListService.upsertTaskList(taskList);
+    await devices[0].taskListService.upsertTask(taskList.id!, task1);
+    await devices[1].taskListService.upsertTaskList(taskList);
+    await devices[1].taskListService.upsertTask(taskList.id!, task2);
+
+    // two-way merge:
+    await devices[0]
+        .taskListService
+        .mergeCrdtJson(await devices[1].taskListService.crdtToJson());
+    await devices[1]
+        .taskListService
+        .mergeCrdtJson(await devices[0].taskListService.crdtToJson());
+
+    // update title in device 1 and description in device 2:
+    await devices[0]
+        .taskListService
+        .upsertTask(taskList.id!, task1..title = 'task1 updated');
+    await devices[1]
+        .taskListService
+        .upsertTask(taskList.id!, task2..description = 'task2 description');
+
+    await devices[0]
+        .taskListService
+        .mergeCrdtJson(await devices[1].taskListService.crdtToJson());
+    expect((await devices[0].taskListService.allTasks).toSet(), {
+      Task(
```

Comment:
This test can easily fail, when the Task-Constructor changes (e.g. when adding a `created` timestamp). Check for the respective values instead (so compare the description and the title explicitly).